src/Inventory/Asset/NetworkPort.php

```diff
@@ -654,6 +654,12 @@ public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = [])
         $port = $this->current_connection ?? $this->current_port;
         $item = new $itemtype();
 
+        $rulesmatched = new \RuleMatchedLog();
+        $agents_id = $this->agent->fields['id'];
+        if (empty($agents_id)) {
+            $agents_id = 0;
+        }
+
```

Comment:
Why this code has been splitted into two parts? It could probably be placed before ``$inputrulelog = [...];`` code.

Reply:
fixed with last commit